src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -166,17 +175,77 @@ public class AddPackageReferenceCommandRunner : IPackageReferenceCommandRunner
                     .OriginalTargetFrameworks
                     .Where(s => compatibleFrameworks.Contains(NuGetFramework.Parse(s)));
 
-                // If the user did not specify a version then update the version to resolved version
-                UpdatePackageVersionIfNeeded(restorePreviewResult, packageReferenceArgs, userSpecifiedFrameworks);
+                // generate a library dependency with all the metadata like Include, Exlude and SuppressParent
+                var libraryDependency = GenerateLibraryDependency(updatedPackageSpec, packageReferenceArgs, restorePreviewResult, userSpecifiedFrameworks);
 
                 msBuild.AddPackageReferencePerTFM(packageReferenceArgs.ProjectPath,
-                    packageReferenceArgs.PackageDependency,
+                    libraryDependency,
                     compatibleOriginalFrameworks);
             }
 
             return 0;
         }
 
+        private static LibraryDependency GenerateLibraryDependency(
+            PackageSpec project,
+            PackageReferenceArgs packageReferenceArgs,
+            RestoreResultPair restorePreviewResult,
+            IEnumerable<NuGetFramework> UserSpecifiedFrameworks)
+        {
+            // get the package resolved version from restore preview result
+            var resolvedVersion = GetPackageVersionFromRestoreResult(restorePreviewResult, packageReferenceArgs, UserSpecifiedFrameworks);
+
+            // create a path resolver to get nuspec file of the package
+            var pathResolver = new FallbackPackagePathResolver(
+                project.RestoreMetadata.PackagesPath,
+                project.RestoreMetadata.FallbackFolders);
+            var info = pathResolver.GetPackageInfo(packageReferenceArgs.PackageDependency.Id, resolvedVersion);
+            var nuspecFilePath = info?.PathResolver.GetManifestFileName(packageReferenceArgs.PackageDependency.Id, resolvedVersion);
```

Comment:
🐛  The file name needs to be combined with the install path to get the full path.

Reply:
`PathResolver` will take care of combining package install path to get the full path.